swaScraper.py

```diff
@@ -159,39 +165,54 @@ def scrape(args):
 
         if args.seniors:
             outbound_fares = browser.find_elements_by_tag_name("ul")[1]
+        elif args.flightnum:
+            # find li which flight number exists
+            xpath = '//button[contains(@class,\'flight-number\') and contains(.,"{0}")]/ancestor::*[self::li][1]'.format(args.flightnum)
+            outbound_fares = browser.find_element_by_xpath(xpath)
         else:
             outbound_fares = browser.find_elements_by_tag_name("ul")[2]
 
-        outbound_prices = outbound_fares.find_elements_by_class_name("currency_dollars")
+        if args.points:
+            outbound_prices = outbound_fares.find_elements_by_class_name("currency_points")
+            for price in outbound_prices:
+                realprice = price.text.replace(",", "")
+                outbound_array.append(int(realprice))
+        else:
+            outbound_prices = outbound_fares.find_elements_by_class_name("currency_dollars")
+            for price in outbound_prices:
+                realprice = price.text.replace("$", "")
+                outbound_array.append(int(realprice))
 
-        for price in outbound_prices:
-            realprice = price.text.replace("$", "")
-            outbound_array.append(int(realprice))
 
         lowest_outbound_fare = min(outbound_array)
 
         if not args.one_way:
             return_fares = browser.find_elements_by_tag_name("ul")[5]
-            return_prices = return_fares.find_elements_by_class_name("currency_dollars")
-
-            for price in return_prices:
-                realprice = price.text.replace("$", "")
-                return_array.append(int(realprice))
+            if args.points:
```

Comment:
Can you think of a way of re-writing this without the redundancy?

Reply:
Same as above